src/frontend/src/flows/authenticate/index.ts

```diff
@@ -22,33 +22,28 @@ import {
 import { unreachable } from "../../utils/utils";
 import { withRef } from "../../utils/lit-html";
 
-type PageElements = {
-  authorizeButton: Ref<HTMLButtonElement>;
-  registerButton: Ref<HTMLLinkElement>;
+type PageProps = {
+  origin: string;
+  onContinue: (arg: bigint) => void;
```

Comment:
shouldn't `onContinue` return a `Promise<void>` according how you use it?

Reply:
`onContinue` can be anything, but it's not `await`ed or anything anywhere. It's really just an action that happens when the user clicks.